src/bespokelabs/curator/request_processor/openai_batch_request_processor.py

```diff
@@ -452,95 +529,100 @@ async def download_batch_to_generic_responses_file(self, batch: Batch) -> str |
+                    request_creation_times[generic_request.original_row_idx] = (
+                        datetime.datetime.now()
                     )
 
-                    response_message = choices[0]["message"]["content"]
-                    response_message, response_errors = parse_response_message(
-                        response_message, self.prompt_formatter.response_format
-                    )
+            with open(response_file, "w") as f:
+                for raw_response in file_content.text.splitlines():
+                    raw_response = json.loads(raw_response)
+                    request_idx = int(raw_response["custom_id"])
+                    generic_request = generic_request_map[request_idx]
 
-                    generic_response = GenericResponse(
-                        response_message=response_message,
-                        response_errors=response_errors,
-                        raw_response=raw_response,
-                        raw_request=None,
-                        generic_request=generic_request,
-                        created_at=request_creation_times[request_idx],
-                        finished_at=datetime.datetime.now(),
-                        token_usage=token_usage,
-                        response_cost=cost,
-                    )
-                f.write(json.dumps(generic_response.model_dump(), default=str) + "\n")
-        return response_file
+                    # TODO(Ryan): Add more specific error handling
+                    if raw_response["response"]["status_code"] != 200:
+                        logger.warning(
+                            f"Request {generic_request} failed with status code {raw_response['response']['status_code']}"
+                        )
+                        generic_response = GenericResponse(
+                            response_message=None,
+                            response_errors=[
+                                f"Request {generic_request} failed with status code {raw_response['response']['status_code']}"
+                            ],
+                            raw_response=raw_response,
+                            raw_request=None,
+                            generic_request=generic_request,
+                            created_at=request_creation_times[request_idx],
+                            finished_at=datetime.datetime.now(),
+                            token_usage=None,
+                            response_cost=None,
+                        )
+                    else:
+                        response_body = raw_response["response"]["body"]
+                        choices = response_body["choices"]
+                        usage = response_body.get("usage", {})
+
+                        token_usage = TokenUsage(
+                            prompt_tokens=usage.get("prompt_tokens", 0),
+                            completion_tokens=usage.get("completion_tokens", 0),
```

Comment:
do we need to discount by 50% since it's batch?

Reply:
fixed